colorama/ansitowin32.py

```diff
@@ -64,12 +64,16 @@ def __init__(self, wrapped, convert=None, strip=None, autoreset=False):
 
         # should we strip ANSI sequences from our output?
         if strip is None:
-            strip = conversion_supported or (not wrapped.closed and not is_a_tty(wrapped))
+            strip = conversion_supported or (
+                hasattr(wrapped, 'closed') and not wrapped.closed and not
+                is_a_tty(wrapped))
         self.strip = strip
 
         # should we should convert ANSI sequences into win32 calls?
         if convert is None:
-            convert = conversion_supported and not wrapped.closed and is_a_tty(wrapped)
+            convert = (conversion_supported and
```

Comment:
The and and or are ok.
* We want to strip the ANSI codes in two separate cases: either we're using a Windows cmd-like terminal (conversion_supported, ANSI codes are not supported) OR we're not using a tty at all (e.g. output redirection)
* To convert the ANSI codes to Windows API we require two conditions: use a cmd-like terminal AND use a tty (no redirection).